### PR TITLE
Handle pointers

### PR DESCRIPTION
**Pull request by MaximeHeckel**
Maybe there's a better / more elegant way to do this, but this should cover handling pointers for now

Before:
```
~/Documents/Projects/go2flow master
❯ go run main.go
type ObjectMeta {
  name?: string,
  generateName?: string,
  namespace?: string,
  selfLink?: string,
  uid?: types.UID,
  resourceVersion?: string,
  generation?: number,
  creationTimestamp?: metav1.Time,
  deletionTimestamp?: UNKNOWN_EXPR_TYPE,
  deletionGracePeriodSeconds?: UNKNOWN_EXPR_TYPE,
  labels?: {[string]: string},
  annotations?: {[string]: string},
  ownerReferences?: Array<metav1.OwnerReference>,
  initializers?: UNKNOWN_EXPR_TYPE,
  finalizers?: Array<string>,
  clusterName?: string,
}

type Volume {
  name: string,
}

type VolumeSource {
}
```

After:
```
~/Documents/Projects/go2flow handle-pointer*
❯ go run main.go
type ObjectMeta {
  name?: string,
  generateName?: string,
  namespace?: string,
  selfLink?: string,
  uid?: types.UID,
  resourceVersion?: string,
  generation?: number,
  creationTimestamp?: metav1.Time,
  deletionTimestamp?: metav1.Time,
  deletionGracePeriodSeconds?: int64,
  labels?: {[string]: string},
  annotations?: {[string]: string},
  ownerReferences?: Array<metav1.OwnerReference>,
  initializers?: metav1.Initializers,
  finalizers?: Array<string>,
  clusterName?: string,
}

type Volume {
  name: string,
}

type VolumeSource {
}
```

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -78,6 +77,13 @@ func GetTypeInfo(fieldType ast.Expr) string {
 			return typeStr
 		}
 		return flowType
+	// *T
+	case *ast.StarExpr:
+		typeStr := fmt.Sprintf("%s", t.X)
+		if strings.Contains(typeStr, "&") {
```
Comment (MaximeHeckel) [inline main.go]: When printing `t.X` we either get &{ Type } or Type, hence this check, not sure it covers all the cases

Comment (kristiehoward) [reply main.go]: Add recursive call to `GetTypeInfo` , this will cover all of those cases

Comment (MaximeHeckel) [reply main.go]: fixed:

```
~/Documents/Projects/go2flow handle-pointer
❯ go run main.go
type ObjectMeta {
  name?: string,
  generateName?: string,
  namespace?: string,
  selfLink?: string,
  uid?: types.UID,
  resourceVersion?: string,
  generation?: number,
  creationTimestamp?: metav1.Time,
  deletionTimestamp?: metav1.Time,
  deletionGracePeriodSeconds?: number,
  labels?: {[string]: string},
  annotations?: {[string]: string},
  ownerReferences?: Array<metav1.OwnerReference>,
  initializers?: metav1.Initializers,
  finalizers?: Array<string>,
  clusterName?: string,
}

type Volume {
  name: string,
}

type VolumeSource {
}
```

Comment (kristiehoward) [reply main.go]: I think i might be going crazy.... So i guess I handled pointers on master 14 days ago (https://github.com/kristiehoward/go2flow/blob/master/main.go#L63-L66) but I don't see that here... is your code rebased? (also you can just do a branch if you want)

For the strings.replace line, shouldn't this line cover it: https://github.com/kristiehoward/go2flow/pull/1/files#diff-7ddfb3e035b42cd70649cc33393fe32cR72

Or could you be more specific about the before / after of the lines that this is addressing for the kube integration, for example

Comment (MaximeHeckel) [reply main.go]: Oh you're right, I'll rebase properly, maybe update the logic (since `getTypeInfo(t.X)` alone didn't cover all the cases)

**Comment by MaximeHeckel** [comment]
Rebased against your master

**Comment by MaximeHeckel** [comment]
Not relevant since the work has already been done and master seems to output the expected result.
Closing this for now

**Pull request closed without merging**